### PR TITLE
Send coverage to Coveralls

### PR DESCRIPTION
**Pull request by szepeviktor**
Docs: https://github.com/marketplace/actions/coveralls-github-action

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -188,10 +188,12 @@ jobs:
...
 
-      - name: "Codecov.io"
+      - name: "Coveralls"
         env:
-          CODECOV_TOKEN: "${{ secrets.CODECOV_TOKEN }}"
-        run: "bash <(curl -s https://codecov.io/bash) -f tests/tmp/clover.xml"
+          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+           composer require twinh/php-coveralls --dev && \
+           vendor/bin/php-coveralls --verbose --coverage_clover=tests/tmp/clover.xml --json_path=tests/tmp/coveralls-upload.json
```
Comment (ondrejmirtes) [inline .github/workflows/build.yml:196]: Could you try following the custom GItHub Action guide instead: https://github.com/marketplace/actions/coveralls-github-action

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Thanks, it works, but it uses a PHP client, to the guide you linked :)

**Comment by ondrejmirtes** [comment]
Also, you can clean up the current Codecov config.

**Comment by szepeviktor** [comment]
Okay!

Success :)

```
Coverage:  84.51% (20811/24626)
Collect git info
Read environment variables
Dump submitting json file: /home/runner/work/phpstan-src/phpstan-src/tests/tmp/coveralls-upload.json
File size: 2,458.04 kB
Submitting to https://coveralls.io/api/v1/jobs
Finish submitting. status: 200 OK
Accepted Job #226389376.1
You can see the build on https://coveralls.io/jobs/66907835
elapsed time: 2.262 sec memory: 16.00 MB
````

https://coveralls.io/jobs/66907835

File: .github/workflows/build.yml (hunk 1 above)
Comment (szepeviktor) [reply .github/workflows/build.yml:196]: That GitHub Action is for LCOV formatted coverage only.
I've tried that, please see the first commits.

```
[error] "2020-08-27T02:10:55.888Z"  'error from lcovParse: ' 'Failed to parse string'
[error] "2020-08-27T02:10:55.889Z"  'input: ' '<?xml version="1.0" encoding="UTF-8"?>\n' +
```

https://github.com/phpstan/phpstan-src/runs/1034404107

Comment (szepeviktor) [reply .github/workflows/build.yml:196]: > There's an official github action repo by coveralls but it seems to be for js based projects only coverallsapp/github-action.

https://github.com/php-coveralls/php-coveralls/issues/273#issuecomment-533221094

Comment (ondrejmirtes) [reply .github/workflows/build.yml:196]: Yeah, that's why I wasn't successful previously. Thanks.

**Review by ondrejmirtes — APPROVED** [review]

**Comment by szepeviktor** [comment]
Thank you.

**Comment by szepeviktor** [comment]
The badge just lit up :)

kép

**Pull request merged**